lib/LoginStrategy.ts

```diff
@@ -302,7 +303,12 @@ export class LoginStrategy extends BaseStrategy {
         null
       );
     }
-    const userDid = this.didUnification(`did:ethr:${payload.address}`);
+    let userDid;
+    if ((await this.getChainName(this.provider)) === 'ewc') {
```

Comment:
Better to rely on the chainId of the SIWE message I think. It is more intuitive as it allows the DID to be generated solely from the message.
```suggestion
    if (payload.chainId === EWC_CHAIN_ID) {
```
If above is used, `getChainName` can also be removed